src/software/new_geom/segment.h

```diff
@@ -13,83 +10,67 @@ class Segment final
      *
      * @param o new start point of segment
      */
-    void setSegStart(Point o)
-    {
-        start = o;
-    }
+    void setStart(Point o);
 
     /**
      * Gets the start point of the segment
      *
      * @return start point of segment
      */
-    const Point& getSegStart() const
-    {
-        return start;
-    }
+    const Point& getStart() const;
 
     /**
      * Sets the end point of the segment to o
      *
      * @param o new end point of segment
      */
-    void setEnd(Point o)
-    {
-        end = o;
-    }
+    void setEnd(Point o);
 
     /**
      * Gets the end point of the segment
      *
      * @return end point of segment
      */
-    const Point& getEnd() const
-    {
-        return end;
-    }
+    const Point& getEnd() const;
+
+    /**
+     * Gets the length of the segment
+     *
+     * @return length of segment
+     */
+    double length() const;
 
     /**
-     * Creates a degenerate Segment at (0, 0)
+     * Gets the squared length of the segment
+     *
+     * @return squared length of segment
      */
-    inline explicit Segment() {}
+    double lengthSquared() const;
 
     /**
      * Creates a Segment that starts and ends at the given points
      */
     inline explicit Segment(const Point& start, const Point& end) : start(start), end(end)
```

Comment:
Could we put the function implementations in `segment.cpp` file?

Reply:
Also add an EXPECT_THROW test for this

Reply:
I can do that but there's a separate ticket for adding tests to `segment`. Did you want me to just do both in this PR?

Reply:
yes please test it because it was added in this PR

Reply:
Moved `length` and `lengthSquared` implementations and added EXPECT_THROW test.

Reply:
Just to double check, we no longer need an `EXPECT_THROW` test here since we're no longer throwing exceptions?

Reply:
This is correct. EXPECT_THROW has been since removed since we allow degen segments.